### ISSUE TITLE
Error with "non-inline" email attachments

### ISSUE THREAD

**Issue by daveherman71** [bcit-ci/CodeIgniter#4746]
Version 3.1.0

There is an issue which occurs in `/system/libraries/Email.php` on line `1471` when you add attachments to an email and none of them have a `Content-ID`.

When there is a Content-ID specified the resulting message looks something like this:

```
Content-Type: image/jpeg; name="test.jpg"
Content-Disposition: inline;
Content-Transfer-Encoding: base64
Content-ID: <test.jpg@57a0a6e46acfc>

/9j/4AAQSkZJRgABAQEBLAEsAAD/4QCqRXhpZgAATU0AKgAAAAgACQEaAAUAAAABAAAAegEbAAUA
AAABAAAAggEoAAMAAAABAAMAAAExAAIAAAAQAAAAigMBAAUAAAABAAAAmgMDAAEAAAABAAAAAFEQ
AAEAAAABAQAAAFERAAQAAAABAAAuIlESAAQAAAABAAAuIgAAAAAAAczvAAAD6AABzO8AAAPocGFp
```

However, when there us no Content-ID specified (i.e. not an inline attachment) then the message looks something like this:

```
Content-Type: image/jpeg; name="test.jpg"
Content-Disposition: attachment;
Content-Transfer-Encoding: base64
/9j/4AAQSkZJRgABAQEBLAEsAAD/4QCqRXhpZgAATU0AKgAAAAgACQEaAAUAAAABAAAAegEbAAUA
AAABAAAAggEoAAMAAAABAAMAAAExAAIAAAAQAAAAigMBAAUAAAABAAAAmgMDAAEAAAABAAAAAFEQ
AAEAAAABAQAAAFERAAQAAAABAAAuIlESAAQAAAABAAAuIgAAAAAAAczvAAAD6AABzO8AAAPocGFp

```
Take note that now there is no blank line between the attachment headers and content.

Because the blank line is missing the email seems to appear corrupted when read - I do not know why this happens. I have tested with IIS SMTP Server and MS Exchange Servers 2010 and 2013.

Simply adding in the blank line seems to fix the issue so I changed line `1471` from

`.(empty($this->_attachments[$i]['cid']) ? '' : 'Content-ID: <'.$this->_attachments[$i]['cid'].'>'.$this->newline.$this->newline)`

to

`.(empty($this->_attachments[$i]['cid']) ? '' : 'Content-ID: <'.$this->_attachments[$i]['cid'].'>'.$this->newline).$this->newline`

I hope this helps someone.

**Comment by vkeranov** [comment]
But I think this is already fixed in the 3.1-stable branch... Check here:
https://github.com/bcit-ci/CodeIgniter/commit/9b0f5fa0339eb1f74ff06e769d71e5911ba2be06

**Comment by narfbg** [comment]
#4739
